transform/mattermost-analytics/models/marts/product/nps/fct_nps_score.sql

```diff
@@ -0,0 +1,85 @@
+      agg='sum',
+      then_value='promoters',
+      quote_identifiers=False,
+      suffix='_promoters'
+  ) }},
+    {{ dbt_utils.pivot(
+      'user_role',
+      dbt_utils.get_column_values(ref('int_user_nps_score_spined'), 'user_role'),
+      agg='sum',
+      then_value='detractors',
+      quote_identifiers=False,
+      suffix='_detractors'
+  ) }},
+    {{ dbt_utils.pivot(
+      'user_role',
+      dbt_utils.get_column_values(ref('int_user_nps_score_spined'), 'user_role'),
+      agg='sum',
+      then_value='passives',
+      quote_identifiers=False,
+      suffix='_passives'
+  ) }},
+    {{ dbt_utils.pivot(
+      'user_role',
+      dbt_utils.get_column_values(ref('int_user_nps_score_spined'), 'user_role'),
+      agg='sum',
+      then_value='nps_users',
+      quote_identifiers=False,
+      suffix='_nps_users'
+  ) }},
+    {{ dbt_utils.pivot(
+      'user_role',
+      dbt_utils.get_column_values(ref('int_user_nps_score_spined'), 'user_role'),
+      agg='sum',
+      then_value='quarterly_promoters',
+      quote_identifiers=False,
+      suffix='_quarterly_promoters'
+  ) }},
+    {{ dbt_utils.pivot(
+      'user_role',
+      dbt_utils.get_column_values(ref('int_user_nps_score_spined'), 'user_role'),
+      agg='sum',
+      then_value='quarterly_detractors',
+      quote_identifiers=False,
+      suffix='_quarterly_detractors'
+  ) }},
+    {{ dbt_utils.pivot(
+      'user_role',
+      dbt_utils.get_column_values(ref('int_user_nps_score_spined'), 'user_role'),
+      agg='sum',
+      then_value='quarterly_passives',
+      quote_identifiers=False,
+      suffix='_quarterly_passives'
+  ) }},
+    {{ dbt_utils.pivot(
+      'user_role',
+      dbt_utils.get_column_values(ref('int_user_nps_score_spined'), 'user_role'),
+      agg='sum',
+      then_value='quarterly_nps_users',
+      quote_identifiers=False,
+      suffix='_quarterly_nps_users'
+  ) }}
+    FROM
+    {{ ref('int_user_nps_score_spined') }}
+    group by activity_date
+    , server_id
+)
+SELECT a.*,
+    b.server_version AS server_version,
+    a.user_promoters + a.team_admin_promoters + a.system_admin_promoters AS promoters,
+    a.user_detractors + a.team_admin_detractors + a.system_admin_detractors AS detractors,
+    a.user_passives + a.team_admin_passives + a.system_admin_passives AS passives,
+    a.user_nps_users + a.team_admin_nps_users + a.system_admin_nps_users AS nps_users,
+    a.user_quarterly_promoters + a.team_admin_quarterly_promoters + a.system_admin_quarterly_promoters AS quarterly_promoters,
+    a.user_quarterly_detractors + a.team_admin_quarterly_detractors + a.system_admin_quarterly_detractors AS quarterly_detractors,
+    a.user_quarterly_passives + a.team_admin_quarterly_passives + a.system_admin_quarterly_passives AS quarterly_passives,
+    a.user_quarterly_nps_users + a.team_admin_quarterly_nps_users + a.system_admin_quarterly_nps_users AS quarterly_nps_users
```

Comment:
nit: According to [naming conventions](https://mattermost.atlassian.net/wiki/spaces/DATAENG/pages/2422603777/DWH+Naming+Conventions), column name for aggregations should indicate the aggregation. 

Most of the columns are counts. So `promoters` could be renamed to `count_promoters_per_day`, `count_promoters_daily` or even `count_promoters`. 

Another nit: `quarterly` can be misleading and lead to confusion that it counts on current quarter (based on activity date). Perhaps use `_last_90d` or something similar as the postfix. It's already in the documentation, but it can help anyone who's doing a quick check on the fact table.

Reply:
Updated the column names following naming conventions